packages/patterns/src/keys/compareKeys.js

```diff
@@ -133,20 +133,25 @@ export const compareKeys = (left, right) => {
       // rank order.
       // Because the invariants above apply to the elements of the array,
       // they apply to the array as a whole.
+      // @ts-expect-error FIXME narrowed
```

Comment:
👆 I’m fine with a link and dittos if these are intended to commit. (but don’t like `FIXME` because I like `FIXME` and `XXX` to be used as markers for work that needs to be ironed out before a merge.